src/drm.cpp

```diff
@@ -1279,22 +1288,56 @@ void drm_unlock_fbid( struct drm_t *drm, uint32_t fbid )
 }
 
 /* Handle the orientation of the display */
-uint64_t get_drm_effective_orientation()
+	void update_drm_effective_orientation(struct drm_t *drm,  struct connector *conn)
 {
-	switch ( g_drmModeOrientation )
+	drm_screen_type screenType = drm_get_screen_type(drm);
+	if ( screenType == DRM_SCREEN_TYPE_INTERNAL )
```

Comment:
Nit: we can use an early return to avoid growing the indentation too much. Something like:

```c
if ( screenType == DRM_SCREEN_TYPE_INTERNAL ) {
	g_drmEffectiveOrientation = DRM_MODE_ROTATE_0;
	return;
}

switch ( g_drmModeOrientation ) {
…
}
```